src/sentry/web/client_config.py

```diff
@@ -251,6 +252,37 @@ def user_details(self) -> Mapping[str, Any] | None:
             user_details["isSuperuser"] = self.user.is_superuser
         return user_details
 
+    @property
+    def regions(self) -> List[Mapping[str, Any]]:
+        """The regions the current user has presence in"""
+        user = self.user
+        region_names = find_all_region_names()
+        if not region_names or not user or not user.id:
+            return self.available_regions
+
+        memberships = user_service.get_organizations(user_id=user.id)
```

Comment:
It's possible to get the organizations a user belongs on in both region and silo modes without resorting to RPC.  Can we pick the right model using a service abstraction or conditional?

Reply:
Deeper question -- /why are doing extra work to find all the regions a user belongs to?/
the alternative design I keep thinking of is, why not list all regions, from static config, and then just fan out to collect organizations?

Reply:
> /why are doing extra work to find all the regions a user belongs to?/
the alternative design I keep thinking of is, why not list all regions, from static config, and then just fan out to collect organizations?

I was trying to save latency on the frontend. The request for `/organizations` is very high in the render tree and blocks rendering of the entire UI while that data is being fetched. 

> Can we pick the right model using a service abstraction or conditional?

I was trying to avoid adding more silo specific logic, but avoiding the additional RPC in regions is probably best.

Reply:
Roger.  It seems super super unfortunate that `/organizations` is such a heavy costing endpoint.
I guess I can't really peer down this path right now to know if this is a good direction other than to defer to you.  My intuition makes me concerned that this sort of logic and extra dependency on RPC for fetching membership is fragile, I wish it were easier to do 'naive, resilient fan out', but I can accept realities.

Reply:
Updated to fan out to all multi-tenant regions. Users that have membership in single-tenant orgs will fetch from those regions as well. I'll do a follow up pull request to prioritize the current org's region and load other regions in the background.

Reply:
Thank you.